lib/diagnostic.js

```diff
@@ -41,13 +44,23 @@ let scopes = [/* your response here */];
 /*5 */   }
 /*6 */ };
 
+// let deathstar = {
+//   weapon: 'Planet destroying laser'   // alderaan
+// };
+// const goBoom = function (weapon) {
+//   console.log("this refers to ", this); // deathStar.fire?
+// };
+//
+//
+// goBoom.call(deathstar); // deathStar.fire.call(alderaan)?
+
 // Write the code that would call the `deathStar`'s `fire` method in the
 // context of `alderaan`. In other words, we want to *reference* the `fire`
 // method and temporarily bind it (or, change `this`) to `alderaan` before
 // executing it.
 //
 let fireOnAlderaan = function () {
-  // your code here
+  deathStar.fire.call(alderaan);
```

Comment:
👍 
You have to return the result in order for tests to pass.
`return deathStar.fire.call(alderaan);`